### DISCUSSION TITLE
Others algorithm

### DISCUSSION THREAD

**Discussion by PaoloFer** [codinasion-archive/codinasion-dsa#17] [General]
_Hi, i really like this project of coding different algorithm in different programming lenguages, and i wanted to ask what will be others algorithms that will be coded_
### Ideas:
_**i was thinking for example to:**_
_(First algorithms that come to my mind)_
- sorting algorithm
- Pathfinding
- Valid Palindrome

_If we can make in a readme file  a list of algorithm i think would be faster to realize them, even for people that want to contribute._

**Comment by harshraj8843** [comment]
Hey @PaoloFer 

Issues for more algorithms will be raised soon 👍🏻 

---

> _**i was thinking for example to:**_
> _(First algorithms that come to my mind)_
> - sorting algorithm
> - Pathfinding
> - Valid Palindrome

Definitely

---

> _If we can make in a readme file  a list of algorithm i think would be faster to realize them, even for people that want to contribute._

This is an new idea, could you please elaborate it

Means :- 
- the root `README.md` or different `README.md`
- the upcoming algorithms or algorithms already open for contribution

**Reply by PaoloFer** [reply]
Sure, i will do everything, I only need to know if the algorithm that i've said are okey or for the project you would follow a different type of algorithm

**Reply by harshraj8843** [reply]
> Sure, i will do everything, I only need to know if the algorithm that i've said are okey or for the project you would follow a different type of algorithm

This project is for any kind of algorithm 👍🏻 

---

If you are planning to work on any algorithm, please comment down below, so that i can raise issues for the respective algorithm.

Means specific algorithm. Like for sorting (merge sort, heap sort, quick sort, etc...)

Thanks 🤗

**Reply by PaoloFer** [reply]
i would like to start with something more easy to increasi the difficult so i'm going to start with bubble sort and after i will go one with the others algorithm

**Reply by PaoloFer** [reply]
_i think that a good difficult progression could be:_ 
1. bubble sort
2. merge sort
3. quick sort
4. heap sort

_tell me if you like this progression or you want to follow another order_

**Reply by PaoloFer** [reply]
so i will start creating the bubble sort  and Palindrome algorithm.
if you agree with me i will search or i will do  a simple image to explane the algorithm, example and so on like in the liner-search readme

**Reply by harshraj8843** [reply]
> so i will start creating the bubble sort and Palindrome algorithm. if you agree with me i will search or i will do a simple image to explane the algorithm, example and so on like in the liner-search readme

You can start creating 🤗 

Image will be created later no worry 👍🏻

**Reply by PaoloFer** [reply]
o i've created an image, but feel free to delete it.
I've created the py and js algorithm of palindrome, now we have to create the issue, i have right to create the issue or not?

**Reply by harshraj8843** [reply]
> o i've created an image, but feel free to delete it. I've created the py and js algorithm of palindrome, now we have to create the issue, i have right to create the issue or not?

You can create the issue, but i think, it will be better if all the related issues be created at once

**Reply by PaoloFer** [reply]
yes, i was searching for the frontend code but i have not found it.

**Reply by PaoloFer** [reply]
Sorry if write here but i wanted to ask the average time for a pull request review and approval.

**Reply by harshraj8843** [reply]
> Sorry if write here but i wanted to ask the average time for a pull request review and approval.

Sorry for the late, it will be merged soon 👍🏻

**Reply by PaoloFer** [reply]
no problem i refer also to the other repo that there are a lot of open pr. It was only a question. I didn't want to make it look like an order

**Comment by PaoloFer** [comment]
I really like this project of the algorithm and i was thinking to some new features or idea, the thing that i wanted to ask is:
this project has only the goal to share the different algorithm in different lengauges or not. Because in my opinion for example adding a comment section under the algorithm could be a great idea for the user that could have question or problem about the solution.

**Reply by harshraj8843** [reply]
comment section is already available on site 

test : https://codinasion.vercel.app/dsa/linear-search

**Reply by PaoloFer** [reply]
sorry i didn't see it.

**Comment by PaoloFer** [comment]
for example is there a to do list of things that have not yet been realized or not?
Sorry if I'm bothering you so much but the fact is that I would really like to be of help in this project

**Reply by harshraj8843** [reply]
Well @PaoloFer , the fact is that i am planning to make all projects of this organisation public (including website code).

But since we have many PR(s) open to review and merge, i don't have enough time to work on that

I am planning to restructure all the private repo before making them public ( they are kind of mess right now ). 

Some of the private repos of this organisation are :- 

- codinasion-website (includes website front-end code) ( Next Js )
- codinasion-data (includes all data from different repos, stored in json format) (updates every 6 hours)
- codinasion-bot (https://github.com/orgs/codinasion/discussions/11)

---

Till then, add your todo list in https://github.com/orgs/codinasion/projects/8

**Reply by PaoloFer** [reply]
Wow nice to do list, this is my first serious open source projects and i want to give everithing i can do and if is possible learn new thing like the front end that i never use js framework but only html css, javascript and database i will happy

**Comment by PaoloFer** [comment]
I was thinking, why don't you have for example a discord server where contributor and member can comunicate?i (i think the comunication would become faster and easier)

**Reply by harshraj8843** [reply]
Wow that's a great idea ❤️ 

will be created soon 🤗

**Reply by PaoloFer** [reply]
If you want i can help setting up the server and if you need I can also add an open source discord bot that i can create with python

**Reply by PaoloFer** [reply]
one thing I would like to point out is that some reviewers are not active in the sense that has been approved a lot of pr but there are other pr that have yet to be approved for days.

**Reply by harshraj8843** [reply]
> If you want i can help setting up the server and if you need I can also add an open source discord bot that i can create with python

That will be great 👍🏻 

A server has already been created, you can use it (if possible)

https://discord.gg/umvcrCwm47

Thanks for your help 🤗

**Reply by PaoloFer** [reply]
Sure

**Reply by PaoloFer** [reply]
Have you a list of role that you want in the server?

**Reply by harshraj8843** [reply]
> Have you a list of role that you want in the server?

Not sure about that, maybe
- Admin (for discord admin(s))
- Team (for maintainers of different projects)
- Member (for codinasion github members)(can be assigned by bot)
- ...

**Reply by PaoloFer** [reply]
And about the differe t projects how do you want to work with different channels creating tag gor the different topic example:general,ideas ecc

**Reply by harshraj8843** [reply]
I was thinking of something like

- General
  - Welcome (for welcome of new members by bot)
  - General (for general conversations)
  - QNA (for Q&A)
- Projects (conversations about projects)
  - codinasion-website
  - codinasion-dsa
  - codinasion-programme
  - ...
- MISC
  - Ideas (for projects ideas)
  - ...

**Reply by PaoloFer** [reply]
I will work as soon as i can

**Reply by PaoloFer** [reply]
i joined the server but i couldn't change nothing because i've not the permission, can you give me the role to change the server?

**Reply by harshraj8843** [reply]
> i joined the server but i couldn't change nothing because i've not the permission, can you give me the role to change the server?

not showing in member list 👀

**Comment by PrajwalBorkar** [comment]
I'm thinking we should add time & space complexity section in the code or in the readme file wherever possible. I think it will helpful. What u guys think??

**Reply by harshraj8843** [reply]
In DSA we are already adding it, in codinasion-programme ?

**Reply by PrajwalBorkar** [reply]
Yeah, I was thinking of programme specific complexity section which should be included in PR. So contributors have to add that.

**Reply by PrajwalBorkar** [reply]
Or maybe just comment line in every code so while learning it would be helpful.

**Reply by harshraj8843** [reply]
yeah that will be great 👍🏻 

we can add it from now onwards, but what about current programmes 🤔

**Reply by PrajwalBorkar** [reply]
We can add this issue. For current programs, we can do good-first-issue as it is vast. Or members can do it, it will take time.

**Comment by PrajwalBorkar** [comment]
We can do one for eg as template then create a issue for this.
